src/System.Text.Json/porting_guide/README.md

```diff
@@ -0,0 +1,173 @@
+# Guide to Porting from Newtonsoft.Json to System.Text.Json
+
+## Overview
+
+* The `System.Text.Json` APIs that are included in .NET Core 3.0 are designed primarily for performance critical scenarios and hence emphasize runtime performance and reducing allocations over high usability and convenience. Furthermore, the in-box JSON stack does not aim to provide feature parity with `Newtonsoft.Json`, especially since it is in its infancy. That said, we still want to make it easier for those whose needs can be fulfilled by the new APIs and those who are looking for better performance. This guide captures some of the common pitfalls and programming patterns that are useful to note when trying to use the new in-box APIs.
+  - Please help contribute to this porting guide to help others onboard to the new APIs.
+  - For an overview on the objectives, constraints, and design requirements of the new APIs, please see the [roadmap](https://github.com/dotnet/corefx/blob/master/src/System.Text.Json/roadmap/README.md).
+  - If you have any feedback or feature requests, feel free to submit issues: https://github.com/dotnet/corefx/issues
+
+## Common Pitfalls, Patterns, and Tips
+
+### Using Ref Structs
+
+* The low-level `Utf8JsonReader` and `Utf8JsonWriter` are `ref structs` which means they have certain limitations (see https://aka.ms/span-safety). For example, they cannot be stored as a field on a class or struct (other than ref structs). To achieve the desired performance, these types must be `ref structs` since they need to cache the input or output `Span<byte>` (which itself is a ref struct). Furthermore, these types are mutable since they hold state. Hence, you should **pass them by ref** rather than by value. Passing them by value would result in a struct copy and the state changes would not be visible to the caller. This differs from `Newtonsoft.Json` since the `JsonTextReader`/`JsonTextWriter` you might be familiar with are classes.
```

Comment:
> This differs from `Newtonsoft.Json` since the `JsonTextReader`/`JsonTextWriter` you might be familiar with are classes.
May I suggest adding a ``types`` behind `JsonTextReader`/`JsonTextWriter`, I'm not a native speaker, but it looks to me that something is missing in this sentence.